op-program/client/derivation_smoke.go

```diff
@@ -0,0 +1,36 @@
+//go:build smoke_test
+// +build smoke_test
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	cldr "github.com/ethereum-optimism/optimism/op-program/client/driver"
+	"github.com/ethereum-optimism/optimism/op-program/client/l1"
+	"github.com/ethereum-optimism/optimism/op-program/client/l2"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+// runDerivation executes the L2 state transition, given a minimal interface to retrieve data.
+func runDerivation(logger log.Logger, cfg *rollup.Config, l2Cfg *params.ChainConfig, l1Head common.Hash, l2Head common.Hash, l2Claim common.Hash, l2ClaimBlockNum uint64, l1Oracle l1.Oracle, l2Oracle l2.Oracle) error {
+	l1Source := l1.NewOracleL1Client(logger, l1Oracle, l1Head)
+	engineBackend, err := l2.NewOracleBackedL2Chain(logger, l2Oracle, l2Cfg, l2Head)
+	if err != nil {
+		return fmt.Errorf("failed to create oracle-backed L2 chain: %w", err)
+	}
+	l2Source := l2.NewOracleEngine(cfg, logger, engineBackend)
+
+	logger.Info("Starting derivation")
+	d := cldr.NewDriver(logger, cfg, l1Source, l2Source, l2ClaimBlockNum)
+	// in smoke test, we only run 10 steps
+	for i := 0; i < 10; i++ {
```

Comment:
I think you can make 10 as a variable like maximumSteps and put it in two types: one for smoke test with value 10, and another for non-smoke test with value -1.  Therefore runDerivation() can be a single with a loop check that `if step >= maximumSteps and maximumSteps >= 0 { break }`

This will minimize the duplicated code.

Reply:
Fixed in https://github.com/ethstorage/optimism/pull/10/commits/69bc631d0586529a1d923eb7d1d093ac3ba2eacc